### PR TITLE
PLT-2713 Added ability for admins to list users not in any team

### PR DESCRIPTION
**Pull request by hmhealey**
This is needed for the new system console user list

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-2713

#### Checklist
- Added or updated unit tests (required for all new features)
- Added API documentation (https://github.com/mattermost/mattermost-api-reference/pull/172)
- All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api4/user_test.go [hunk 1]
```diff
@@ -690,6 +690,57 @@ func TestGetUsers(t *testing.T) {
...
+		Username: "a000000001" + model.NewId(),
+		Email:    "success+" + model.NewId() + "@simulator.amazonses.com",
+		Password: "Password1",
+	})
+	CheckNoError(t, resp)
+	defer app.Srv.Store.User().PermanentDelete(user2.Id)
+
+	if rusers, resp := SystemAdminClient.GetUsersWithoutTeam(0, 100, ""); resp.Error != nil {
+		t.Fatal(resp.Error)
+	} else {
```
Comment (jwilander) [inline api4/user_test.go]: Can you use the pattern like this for the test?
```
<your-result>, resp := <your-driver-call>
CheckNoError(t, resp)

// do your result validation here
```

I'm trying to keep the pattern consistent through api4 tests and it also removes a level of nesting :)

**Comment by enahum** [comment]
@hmhealey needs rebase

File: api4/user_test.go (hunk 1 above)
Comment (hmhealey) [inline api4/user_test.go]: Yep, will do

**Comment by hmhealey** [comment]
@jwilander Fixed finally

**Pull request merged**